src/mjolnir/ingest_transit.cc

```diff
@@ -162,66 +163,57 @@ std::priority_queue<tileTransitInfo> select_transit_tiles(const boost::property_
       LOG_INFO("Done Reading Feed");
       const auto& stops = feed.get_stops();
 
+      // 1st pass to add all the stations, so we can add stops to its children in a 2nd pass
       for (const auto& stop : stops) {
-        if (stop.stop_id.empty()) {
-          continue;
+        if (stop.location_type == gtfs::StopLocationType::Station) {
+          auto& tile_info = get_tile_info(stop);
+          tile_info.stations.insert({stop.stop_id, feed_path});
         }
-        gtfs::Id currStopId = stop.stop_id;
-        float x = stop.stop_lon;
-        float y = stop.stop_lat;
-
-        GraphId currId =
-            GraphId(tile_level.tiles.TileId(y, x), TileHierarchy::GetTransitLevel().level, 0);
-
-        tileTransitInfo currTile;
+      }
 
-        if (tile_map.find(currId) != tile_map.end()) {
-          currTile = tile_map[currId];
-        } else {
-          currTile.graphId = currId;
+      // 2nd pass to add the platforms/stops
+      for (const auto& stop : stops) {
+        // TODO: GenericNode & BoardingArea could be useful at some point
+        if (!(stop.location_type == gtfs::StopLocationType::StopOrPlatform) &&
+            !(stop.location_type == gtfs::StopLocationType::EntranceExit)) {
+          continue;
         }
 
-        if (stop.location_type == gtfs::StopLocationType::StopOrPlatform ||
-            stop.location_type == gtfs::StopLocationType::EntranceExit) {
-          auto parent_in_tile = currTile.tile_stops.find({stop.parent_station, feed_path});
-          if (parent_in_tile != currTile.tile_stops.end()) {
-            currTile.tile_station_children.insert({{stop.parent_station, feed_path}, stop.stop_id});
-          } else {
-            currTile.tile_stops.insert({stop.stop_id, feed_path});
-          }
+        auto& tile_info = get_tile_info(stop);
+
+        // if this station doesn't exist, we need to create it: we use the fact that this entry is
+        // not a station type to fake a station object in write_stops()
+        auto station_in_tile = tile_info.stations.find({stop.parent_station, feed_path});
+        if (station_in_tile != tile_info.stations.end()) {
+          tile_info.station_children.insert({{stop.parent_station, feed_path}, stop.stop_id});
         } else {
-          currTile.tile_stops.insert({stop.stop_id, feed_path});
+          // we don't have the parent station, if
+          // 1) this stop has none or 2) its parent station is in another tile
+          // TODO: need to handle the 2nd case somehow!
+          tile_info.stations.insert({stop.stop_id, feed_path});
+          tile_info.station_children.insert({{stop.stop_id, feed_path}, stop.stop_id});
         }
```

Comment:
if the station is in this tile, we can add this one to the children of it. if it doesn't have a parent station at all, we add it to both, the `stations` & its `children`. further down we will distinguish, it won't actually be added as a station.

however, we need to somehow handle the case where the parent station is not even in the same tile as a platform, which surely happens somewhere in the world.

Reply:
yeah in the builder or some other part of the code we mention this and i did cover it with @chris-jpark. it just didnt seem like something we should worry about on the first go but definitely will hit it in a real feed. if we can detect its happening (which we can) we should log about it. also we should make a test case for it eventually too. but yeah first things first lets get it working haha